x/pool-incentives/keeper/grpc_query.go

```diff
@@ -143,52 +142,41 @@ func (q Querier) IncentivizedPools(ctx context.Context, _ *types.QueryIncentiviz
 	// While there are exceptions, typically the number of incentivizedPools
 	// equals to the number of incentivized gauges / number of lockable durations.
 	incentivizedPools := make([]types.IncentivizedPool, 0, len(distrInfo.Records)/len(lockableDurations))
-	incentivizedPoolIDs := make(map[uint64]time.Duration)
 
 	// Loop over the distribution records and fill in the incentivized pools struct.
+recordsLoop:
 	for _, record := range distrInfo.Records {
 		for _, lockableDuration := range lockableDurations {
 			poolId, err := q.Keeper.GetPoolIdFromGaugeId(sdkCtx, record.GaugeId, lockableDuration)
-			if err == nil {
+			// If we cannot find the pool id, it's possible this gauge Id is a group gauge.
+			if err != nil {
+				group, err := q.Keeper.incentivesKeeper.GetGroupByGaugeID(sdkCtx, record.GaugeId)
+				if err != nil {
+					continue recordsLoop
+				}
+				poolIds, durations, err := q.Keeper.incentivesKeeper.GetPoolIdsAndDurationsFromGroup(sdkCtx, group)
+				if err != nil {
+					continue recordsLoop
+				}
+				for i, poolId := range poolIds {
+					incentivizedPool := types.IncentivizedPool{
+						PoolId:           poolId,
+						LockableDuration: durations[i],
+						GaugeId:          record.GaugeId,
+					}
+
+					incentivizedPools = append(incentivizedPools, incentivizedPool)
+				}
+				// no need to loop over the remaining lockableDurations so we move
+				// on to the next record.
+				continue recordsLoop
+			} else {
 				incentivizedPool := types.IncentivizedPool{
 					PoolId:           poolId,
 					LockableDuration: lockableDuration,
 					GaugeId:          record.GaugeId,
 				}
 
-				incentivizedPools = append(incentivizedPools, incentivizedPool)
-				incentivizedPoolIDs[poolId] = lockableDuration
-			}
-		}
-	}
-
-	// Only run the following if the above loop determined there were incentivized pools.
-	if len(incentivizedPoolIDs) > 0 {
-		// Retrieve the migration records between balancer pools and concentrated liquidity pools.
-		// This comes from the superfluid keeper, since superfluid is the only pool incentives connected
-		// module that has access to the gamm modules store.
-		migrationRecords, err := q.gammKeeper.GetAllMigrationInfo(sdkCtx)
-		if err != nil {
-			return nil, status.Error(codes.Internal, err.Error())
-		}
-
-		// Iterate over all migration records.
-		for _, record := range migrationRecords.BalancerToConcentratedPoolLinks {
-			// If the cl pool is not in the list of incentivized pools, skip it.
-			lockableDuration, incentivized := incentivizedPoolIDs[record.ClPoolId]
-			if !incentivized {
-				continue
-			}
-
-			// Add the indirectly incentivized balancer pools to the list of incentivized pools.
-			gaugeId, err := q.Keeper.GetPoolGaugeId(sdkCtx, record.BalancerPoolId, lockableDuration)
-			if err == nil {
-				incentivizedPool := types.IncentivizedPool{
-					PoolId:           record.BalancerPoolId,
-					LockableDuration: lockableDuration,
-					GaugeId:          gaugeId,
-				}
-
```

Comment:
Note to reviewer: This is now removed since we don't incentivize pools indirectly through pool links anymore.